lib/controllers/src/Current.cpp

```diff
@@ -2,39 +2,40 @@
 
 using namespace Controller;
 
-Current::Current(Actuator::Actuator &actuator, Encoder::Encoder &encoder,
-                 const std::optional<std::reference_wrapper<Sensor::CurrentSensor> const> &currentSensor, PID::PID &pid,
-                 float maxDegPerSec, float maxCurrent, PinName lowerLimit, PinName upperLimit,
+Current::Current(Actuator::Actuator &actuator, Encoder::Encoder &encoder, Sensor::CurrentSensor &currentSensor,
+                 PID::PID &pid, float maxDegPerSec, float maxCurrent, PinName lowerLimit, PinName upperLimit,
                  bool ignoreDegPerSecChecks, bool ignoreCurrentChecks, bool ignoreLimitSwitchChecks)
     : ActuatorController(actuator, encoder, currentSensor, maxDegPerSec, maxCurrent, lowerLimit, upperLimit,
                          ignoreDegPerSecChecks, ignoreCurrentChecks, ignoreLimitSwitchChecks),
-      m_pid(pid) {}
+      m_pid(pid) {
+  //m_currentSensor.value() = currentSensor;
+}
 
 bool Current::update() {
+  /* Proceed even if encoder read fails, but report */
+  bool enc_update_success = m_encoder.update();
+  bool cs_update_success  = m_currentSensor.value().get().update();
+  bool stop_required      = false;
   if (shouldStop()) {
     stop();
+    stop_required = true;
   } else {
-    float current = 0;
-    if (m_currentSensor && m_currentSensor.value().get().read(current)) {
-      m_actuator.setValue(m_pid.compute(m_setpoint.load(), current));
-      return true;
-    }
+    m_actuator.setValue(m_pid.compute(m_setpoint.load(), m_currentSensor.value().get().read()));
   }
-  return false;
+  return enc_update_success && cs_update_success && !stop_required;
```

Comment:
we return false if we there's a failure or if we had to stop. the stop() is hit if our plant is saturated and we wanna move in the direction of saturation. seem fair?